save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/KotlinUtils.kt

```diff
@@ -0,0 +1,10 @@
+/**
+ * This file contains util methods for kotlin
+ */
+
+package com.saveourtool.save.utils
+
+/**
+ * @return current value cast to [R] or null if casting is not possible
+ */
+inline fun <T : Any, reified R : Any> T.castOrNull(): R? = takeIf { it is R }?.let { it as R }
```

Comment:
Does it do the same as `as?` operator?

Reply:
yeap, read documentation, but missed that: https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator